javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/SitewiseActions.java

```diff
@@ -0,0 +1,479 @@
+import software.amazon.awssdk.services.iotsitewise.model.DeletePortalResponse;
+import software.amazon.awssdk.services.iotsitewise.model.DescribePortalRequest;
+import software.amazon.awssdk.services.iotsitewise.model.GatewayPlatform;
+import software.amazon.awssdk.services.iotsitewise.model.GetAssetPropertyValueRequest;
+import software.amazon.awssdk.services.iotsitewise.model.GetAssetPropertyValueResponse;
+import software.amazon.awssdk.services.iotsitewise.model.GreengrassV2;
+import software.amazon.awssdk.services.iotsitewise.model.ListAssetModelPropertiesRequest;
+import software.amazon.awssdk.services.iotsitewise.model.ListAssetModelsRequest;
+import software.amazon.awssdk.services.iotsitewise.model.Measurement;
+import software.amazon.awssdk.services.iotsitewise.model.PropertyDataType;
+import software.amazon.awssdk.services.iotsitewise.model.PropertyType;
+import software.amazon.awssdk.services.iotsitewise.model.PutAssetPropertyValueEntry;
+import software.amazon.awssdk.services.iotsitewise.model.TimeInNanos;
+import software.amazon.awssdk.services.iotsitewise.model.Variant;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.stream.Collectors;
+
+// snippet-start:[iotsitewise.java2.actions.main]
+public class SitewiseActions {
+
+    private static final Logger logger = LoggerFactory.getLogger(SitewiseActions.class);
+
+    private static IoTSiteWiseAsyncClient ioTSiteWiseAsyncClient;
+
+    private static IoTSiteWiseAsyncClient getAsyncClient() {
+        if (ioTSiteWiseAsyncClient == null) {
+            SdkAsyncHttpClient httpClient = NettyNioAsyncHttpClient.builder()
+                .maxConcurrency(100)
+                .connectionTimeout(Duration.ofSeconds(60))
+                .readTimeout(Duration.ofSeconds(60))
+                .writeTimeout(Duration.ofSeconds(60))
+                .build();
+
+            ClientOverrideConfiguration overrideConfig = ClientOverrideConfiguration.builder()
+                .apiCallTimeout(Duration.ofMinutes(2))
+                .apiCallAttemptTimeout(Duration.ofSeconds(90))
+                .retryStrategy(RetryMode.STANDARD)
+                .build();
+
+            ioTSiteWiseAsyncClient = IoTSiteWiseAsyncClient.builder()
+                .httpClient(httpClient)
+                .overrideConfiguration(overrideConfig)
+                .build();
+        }
+        return ioTSiteWiseAsyncClient;
+    }
+
+    // snippet-start:[sitewise.java2_create_asset_model.main]
+    /**
+     * Creates an asset model asynchronously.
+     *
+     * @param name the name of the asset model to create
+     * @return a {@link CompletableFuture} that completes with the created {@link CreateAssetModelResponse} when the operation is complete
+     */
```

Comment:
Throws `RuntimeException` not documented. That's another reason why I don't think we really want to do rethrow an exception if we don't need to.

Reply:
fixed